service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/AwaitCloneControlledAzureStorageContainerResourceFlightStep.java

```diff
@@ -0,0 +1,103 @@
+
+import static bio.terra.workspace.common.utils.FlightUtils.FLIGHT_POLL_CYCLES;
+import static bio.terra.workspace.common.utils.FlightUtils.FLIGHT_POLL_SECONDS;
+import static bio.terra.workspace.common.utils.FlightUtils.validateRequiredEntries;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.FlightState;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.DatabaseOperationException;
+import bio.terra.stairway.exception.FlightWaitTimedOutException;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.FlightUtils;
+import bio.terra.workspace.generated.model.ApiClonedControlledAzureStorageContainer;
+import bio.terra.workspace.generated.model.ApiCreatedControlledAzureStorageContainer;
+import bio.terra.workspace.service.job.JobMapKeys;
+import bio.terra.workspace.service.resource.controlled.cloud.azure.storageContainer.ControlledAzureStorageContainerResource;
+import bio.terra.workspace.service.resource.model.StewardshipType;
+import bio.terra.workspace.service.resource.model.WsmResourceType;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys;
+import bio.terra.workspace.service.workspace.model.WsmCloneResourceResult;
+import bio.terra.workspace.service.workspace.model.WsmResourceCloneDetails;
+import com.fasterxml.jackson.core.type.TypeReference;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Wait for the clone storage container flight to complete and add the result to the appropriate map
+ */
+public class AwaitCloneControlledAzureStorageContainerResourceFlightStep implements Step {
+
+  private final ControlledAzureStorageContainerResource resource;
+  private final String subflightId;
+
+  public AwaitCloneControlledAzureStorageContainerResourceFlightStep(
+      ControlledAzureStorageContainerResource resource, String subflightId) {
+    this.resource = resource;
+    this.subflightId = subflightId;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    // wait for the flight
+    try {
+      FlightState subflightState =
+          context.getStairway().waitForFlight(subflightId, FLIGHT_POLL_SECONDS, FLIGHT_POLL_CYCLES);
+      WsmResourceCloneDetails cloneDetails = new WsmResourceCloneDetails();
+      WsmCloneResourceResult cloneResult =
+          WorkspaceCloneUtils.flightStatusToCloneResult(subflightState.getFlightStatus(), resource);
+      cloneDetails.setResult(cloneResult);
+
+      FlightMap resultMap = FlightUtils.getResultMapRequired(subflightState);
+      var clonedContainer =
+          resultMap.get(
+              JobMapKeys.RESPONSE.getKeyName(), ApiClonedControlledAzureStorageContainer.class);
```

Comment:
I think this is the wrong type to cast to in my testing, you'll want `ClonedAzureStorageContainer`